client/src/data/reducers/authReducers.ts

```diff
@@ -0,0 +1,43 @@
+import { LOGIN_SUCCESS, LOGIN_FAILURE, LOGOUT } from "../actions/actionTypes";
+import { User } from "../types/userTypes";
+
+interface userState {
+  loggedIn: boolean,
+  token: string;
+  user: User | null;
+}
+
+export interface userObject {
+  user: userState,
+}
+
+const initialState: userState = {
+  loggedIn: false,
+  token: "",
+  user: null
+};
+
+export default function (
+  state: userState = initialState,
+  action: { type: string; payload: any }
+) {
+  switch (action.type) {
+    case LOGIN_SUCCESS:
+      // todo: type the payload when registration and login are merged
+      const user = action.payload.data && action.payload.data.user ? action.payload.data.user : null;
+      const token = action.payload.data && action.payload.data.token ? action.payload.data.token : "";
+
+      return {
+        ...state,
+        loggedIn: true,
+        token,
+        user
+      };
+    case LOGIN_FAILURE:
+      return {};
```

Comment:
Would it be safer to return objects for `LOGIN_FAILURE` and `LOGOUT` that have filled fields?

Reply:
hm, with either case we expect there to be no state. i returned nothing so we don't accidentally persist anything state store. what should we return?

Reply:
I was thinking we could return the initial state so that if anything were to try to access the fields, they would get a value that's more meaningful than undefined. not very consequential tho, so I'll leave it up to u
```
{
  logggedIn: false,
  token: "",
  user: null
}
```

Reply:
that's a good point